lib/msal-browser/docs/iframe-usage.md

```diff
@@ -0,0 +1,140 @@
+# Using MSAL in iframed apps
+
+By default, MSAL prevents full-frame redirects to **Azure AD** login UX when an app is rendered inside an iframe, which means you cannot use [redirect APIs](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/initialization.md#redirect-apis) for user interaction with the IdP:
+
+- This restriction is imposed since **Azure AD** will refuse to render prompts for **credential entry** and **consent** in an iframe by throwing the `X-FRAME OPTIONS SET TO DENY` error (see: [x-frame options](https://html.spec.whatwg.org/multipage/browsing-the-web.html#the-x-frame-options-header)), a measure taken to prevent [clickjacking attacks](https://owasp.org/www-community/attacks/Clickjacking).
+- Instead, you'll have to rely on MSAL's [popup APIs](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/initialization.md#popup-apis) if user interaction is required, and/or silent APIs (`ssoSilent()`, `acquireTokenSilent()`) if user interaction can be avoided.
+- When using [popup APIs](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/initialization.md#popup-apis), you need to take into account any [sandboxing](https://html.spec.whatwg.org/multipage/origin.html#sandboxing) restrictions imposed by the parent app. In particular, the parent app needs to set `allow-popups` flag when the iframe is sandboxed.
+
+**Azure AD B2C** offers an [embedded sign-in experience](https://docs.microsoft.com/azure/active-directory-b2c/embedded-login), which allows rendering a custom login UX in an iframe. Since MSAL prevents redirect in iframes by default, you'll need to set the [allowRedirectInIframe](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/configuration.md#system-config-options) configuration option to **true** in order to make use of this feature. Note that enabling this option for apps on **Azure AD** is not recommended, due to the above restriction.
+
+## Browser restrictions
+
+Because Azure AD session cookies within an iframe are considered [3rd party cookies](https://developer.mozilla.org/en-US/docs/Web/HTTP/Cookies#third-party_cookies), certain browsers (for example **Safari** or **Chrome** in *incognito* mode with 3rd party cookies disabled) either block or clear these cookies. This will affect the **single sign-on** experience for iframed apps as they will not have access to IdP's session cookies (see: [Single sign-on](#single-sign-on)).
```

Comment:
```suggestion
Because Azure AD session cookies within an iframe are considered [3rd party cookies](https://developer.mozilla.org/docs/Web/HTTP/Cookies#third-party_cookies), certain browsers (for example **Safari** or **Chrome** in *incognito* mode with 3rd party cookies disabled) either block or clear these cookies. This will affect the **single sign-on** experience for iframed apps as they will not have access to IdP's session cookies (see: [Single sign-on](#single-sign-on)).
```